identity/aziot-identityd/src/identity.rs

```diff
@@ -719,6 +719,14 @@ impl IdentityManager {
         assert!(!status.eq_ignore_ascii_case("assigning"));
```

Comment:
This should also be an `if` + `return Err` rather than a crashing `assert!`

Reply:
I think Rajeev made it an assert because `dps_client.register` above doesn't return unless the status is not `assigning`. So it's expected to never be `assigning`. I'll remove the check though, it doesn't do much.

Reply:
Ah, in that case I'd rather keep the assert actually, and also add a comment with that explanation.

Reply:
Ok, fixed.